In [57]:
import pandas as pd
import numpy as np

In [58]:
df=pd.read_csv(r"C:\Users\Aswini\Downloads\Data Ananlysis-20210717T051626Z-001\Data Ananlysis\food.txt",sep="\t")

In [59]:
df.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [60]:
df.tail(7)

order_id  quantity            item_name  \
4615      1832         1   Chicken Soft Tacos   
4616      1832         1  Chips and Guacamole   
4617      1833         1        Steak Burrito   
4618      1833         1        Steak Burrito   
4619      1834         1   Chicken Salad Bowl   
4620      1834         1   Chicken Salad Bowl   
4621      1834         1   Chicken Salad Bowl   

                                     choice_description item_price  
4615   [Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]     $8.75   
4616                                                NaN     $4.45   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75

In [61]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


In [62]:
df.shape

(4622, 5)

In [63]:
df[df['choice_description'].isnull()==True]

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
6            3         1                          Side of Chips   
10           5         1                    Chips and Guacamole   
14           7         1                    Chips and Guacamole   
...        ...       ...                                    ...   
4600      1827         1                    Chips and Guacamole   
4605      1828         1                    Chips and Guacamole   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4616      1832         1                    Chips and Guacamole   

     choice_description item_price  
0                   NaN     $2.39   
3                   NaN     $2.39   
6                   NaN     $1.69   
10                  NaN     $4.45   
14                  NaN     $4.45   
...                 ...        ...  
4600                NaN     $4.45   
4605                NaN     $4.45   
4613                NaN     $2.15   
4614                NaN     $1.50   
4616                NaN     $4.45   

[1246 rows x 5 columns]

#### there are 1246 no of nul values in choice description columns

In [21]:
#names of all columns

df.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [64]:
df.index

RangeIndex(start=0, stop=4622, step=1)

In [65]:
df.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [66]:
#which was the most ordered item and how many times it was ordered

In [34]:
c=df.groupby(['item_name']).sum()
c=c.sort_values(by=['quantity'],ascending=False)
c.head()

order_id  quantity
item_name                              
Chicken Bowl           713926       761
Chicken Burrito        497303       591
Chips and Guacamole    449959       506
Steak Burrito          328437       386
Canned Soft Drink      304753       351

In [35]:
#What was the most ordered item in the choice_description column?

In [36]:
df.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

In [39]:
a=df.groupby(['choice_description']).sum()
a.head()

order_id  quantity
choice_description                                                    
[Adobo-Marinated and Grilled Chicken, Pinto Bea...       214         1
[Adobo-Marinated and Grilled Chicken, [Sour Cre...       279         1
[Adobo-Marinated and Grilled Chicken]                    217         1
[Adobo-Marinated and Grilled Steak, [Sour Cream...       279         1
[Adobo-Marinated and Grilled Steak]                      217         1

In [43]:
a.sort_values(by=['quantity'],ascending=False).head(1)

order_id  quantity
choice_description                    
[Diet Coke]           123455       159

In [45]:
#Turn the item price into a float
df.head(3)

order_id  quantity                     item_name choice_description  \
0         1         1  Chips and Fresh Tomato Salsa                NaN   
1         1         1                          Izze       [Clementine]   
2         1         1              Nantucket Nectar            [Apple]   

  item_price  
0     $2.39   
1     $3.39   
2     $3.39

In [46]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


In [73]:
dollar=lambda x:float(x[1:-1])


df['item_price']=df['item_price'].apply(dollar)

In [74]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            4622 non-null   int64  
 1   quantity            4622 non-null   int64  
 2   item_name           4622 non-null   object 
 3   choice_description  3376 non-null   object 
 4   item_price          4622 non-null   float64
 5   revenue             4622 non-null   object 
dtypes: float64(1), int64(2), object(3)
memory usage: 216.8+ KB


In [51]:
#How much was the revenue for the period in the dataset?

In [52]:
df.head(3)

order_id  quantity                     item_name choice_description  \
0         1         1  Chips and Fresh Tomato Salsa                NaN   
1         1         1                          Izze       [Clementine]   
2         1         1              Nantucket Nectar            [Apple]   

   item_price  
0        2.39  
1        3.39  
2        3.39

In [79]:
revenue=(df['item_price']*df['quantity']).sum()
print('the revenue for that period was ','$',revenue)


the revenue for that period was  $ 39237.02


In [80]:
#print a data frame with only two columns item_name and item_price

In [93]:
df.drop(['revenue'],axis=1,inplace=True)

In [87]:
df[['item_name','item_price']]

item_name  item_price
0              Chips and Fresh Tomato Salsa        2.39
1                                      Izze        3.39
2                          Nantucket Nectar        3.39
3     Chips and Tomatillo-Green Chili Salsa        2.39
4                              Chicken Bowl       16.98
...                                     ...         ...
4617                          Steak Burrito       11.75
4618                          Steak Burrito       11.75
4619                     Chicken Salad Bowl       11.25
4620                     Chicken Salad Bowl        8.75
4621                     Chicken Salad Bowl        8.75

[4622 rows x 2 columns]

In [88]:
#deleting dulicate item price and item name


In [94]:
filtered_df=df.drop_duplicates(['item_name','item_price'])
filtered_df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4237      1691         2                    Chips and Guacamole   
4354      1738         2                       Steak Soft Tacos   
4489      1786         4                    Chips and Guacamole   
4509      1793         1                                  Chips   
4510      1793         1                          Barbacoa Bowl   

                                     choice_description  item_price  
0                                                   NaN        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                   NaN        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
...                                                 ...         ...  
4237                                                NaN        8.50  
4354            [Fresh Tomato Salsa, [Cheese, Lettuce]]       18.50  
4489                                                NaN       17.80  
4509                                                NaN        1.99  
4510                                        [Guacamole]       11.49  

[209 rows x 5 columns]

In [97]:
#selecting the products with quantity only 1
one_prod=filtered_df[filtered_df['quantity']==1]
one_prod

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
5            3         1                           Chicken Bowl   
...        ...       ...                                    ...   
3750      1500         1                         Carnitas Salad   
3989      1598         1                          Chicken Salad   
4235      1691         1                           Chicken Bowl   
4509      1793         1                                  Chips   
4510      1793         1                          Barbacoa Bowl   

                                     choice_description  item_price  
0                                                   NaN        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                   NaN        2.39  
5     [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...       10.98  
...                                                 ...         ...  
3750  [[Fresh Tomato Salsa (Mild), Roasted Chili Cor...        8.99  
3989  [Roasted Chili Corn (Medium), [Lettuce, Black ...        8.19  
4235                                       [White Rice]        8.50  
4509                                                NaN        1.99  
4510                                        [Guacamole]       11.49  

[135 rows x 5 columns]

In [101]:
one_prod[['item_name','item_price']].sort_values(by=['item_price'],ascending=False)

item_name  item_price
606        Steak Salad Bowl       11.89
1229    Barbacoa Salad Bowl       11.89
1132    Carnitas Salad Bowl       11.89
93         Carnitas Burrito       11.75
1008  Carnitas Crispy Tacos       11.75
...                     ...         ...
6             Side of Chips        1.69
329           Bottled Water        1.50
263       Canned Soft Drink        1.25
28              Canned Soda        1.09
34            Bottled Water        1.09

[135 rows x 2 columns]

In [102]:
#What was the quantity of the most expensive item ordered?
costliest_item=df.sort_values(by=['item_price'],ascending=False)
costliest_item

order_id  quantity                     item_name  \
3598      1443        15  Chips and Fresh Tomato Salsa   
3480      1398         3                 Carnitas Bowl   
1254       511         4               Chicken Burrito   
3602      1443         4               Chicken Burrito   
3601      1443         3                Veggie Burrito   
...        ...       ...                           ...   
3936      1578         1                   Canned Soda   
2922      1162         1                 Bottled Water   
1396       567         1                   Canned Soda   
2562      1014         1                   Canned Soda   
1457       591         1                   Canned Soda   

                                     choice_description  item_price  
3598                                                NaN       44.25  
3480  [Roasted Chili Corn Salsa, [Fajita Vegetables,...       35.25  
1254  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       35.00  
3602  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...       35.00  
3601  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...       33.75  
...                                                 ...         ...  
3936                                  [Diet Dr. Pepper]        1.09  
2922                                                NaN        1.09  
1396                                        [Coca Cola]        1.09  
2562                                        [Coca Cola]        1.09  
1457                                           [Sprite]        1.09  

[4622 rows x 5 columns]

In [110]:
#How many times were a Veggie Salad Bowl ordered?
veggie_salad=df[df['item_name']=='Veggie Salad Bowl'].count()
veggie_salad

order_id              18
quantity              18
item_name             18
choice_description    18
item_price            18
dtype: int64

In [107]:
df

order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price  
0                                                   NaN        2.39  
1                                          [Clementine]        3.39  
2                                               [Apple]        3.39  
3                                                   NaN        2.39  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98  
...                                                 ...         ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75  

[4622 rows x 5 columns]

In [111]:
df=pd.read_csv(r"https://raw.githubusercontent.com/justmarkham/pandas-videos/master/data/drinks.csv",sep=",")
df.head()

country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   
3      Andorra            245              138            312   
4       Angola            217               57             45   

   total_litres_of_pure_alcohol continent  
0                           0.0      Asia  
1                           4.9    Europe  
2                           0.7    Africa  
3                          12.4    Europe  
4                           5.9    Africa

In [112]:
#Which continent drinks more beer on average?

In [115]:
avg_beer_consumption=df.groupby(['continent']).mean().sort_values(by='beer_servings',ascending=False)
avg_beer_consumption

beer_servings  spirit_servings  wine_servings  \
continent                                                      
Europe            193.777778       132.555556     142.222222   
South America     175.083333       114.750000      62.416667   
North America     145.434783       165.739130      24.521739   
Oceania            89.687500        58.437500      35.625000   
Africa             61.471698        16.339623      16.264151   
Asia               37.045455        60.840909       9.068182   

               total_litres_of_pure_alcohol  
continent                                    
Europe                             8.617778  
South America                      6.308333  
North America                      5.995652  
Oceania                            3.381250  
Africa                             3.007547  
Asia                               2.170455

In [118]:
#For each continent print the statistics for wine consumption.
statistics=df.groupby(by='continent').describe()
wine_statistics=statistics['wine_servings']
wine_statistics

count        mean        std  min   25%    50%     75%    max
continent                                                                   
Africa          53.0   16.264151  38.846419  0.0   1.0    2.0   13.00  233.0
Asia            44.0    9.068182  21.667034  0.0   0.0    1.0    8.00  123.0
Europe          45.0  142.222222  97.421738  0.0  59.0  128.0  195.00  370.0
North America   23.0   24.521739  28.266378  1.0   5.0   11.0   34.00  100.0
Oceania         16.0   35.625000  64.555790  0.0   1.0    8.5   23.25  212.0
South America   12.0   62.416667  88.620189  1.0   3.0   12.0   98.50  221.0

In [119]:
wine_statistics=df.groupby(by='continent')['wine_servings'].describe()

wine_statistics

count        mean        std  min   25%    50%     75%    max
continent                                                                   
Africa          53.0   16.264151  38.846419  0.0   1.0    2.0   13.00  233.0
Asia            44.0    9.068182  21.667034  0.0   0.0    1.0    8.00  123.0
Europe          45.0  142.222222  97.421738  0.0  59.0  128.0  195.00  370.0
North America   23.0   24.521739  28.266378  1.0   5.0   11.0   34.00  100.0
Oceania         16.0   35.625000  64.555790  0.0   1.0    8.5   23.25  212.0
South America   12.0   62.416667  88.620189  1.0   3.0   12.0   98.50  221.0

In [125]:
#country consuming maximum amount of beer
beer_quantity=df.groupby(['country']).sum().sort_values('beer_servings',ascending=False)
beer_quantity

beer_servings  spirit_servings  wine_servings  \
country                                                         
Namibia                   376                3              1   
Czech Republic            361              170            134   
Gabon                     347               98             59   
Germany                   346              117            175   
Lithuania                 343              244             56   
...                       ...              ...            ...   
Kuwait                      0                0              0   
Iran                        0                0              0   
Cook Islands                0              254             74   
Bangladesh                  0                0              0   
Libya                       0                0              0   

                total_litres_of_pure_alcohol  
country                                       
Namibia                                  6.8  
Czech Republic                          11.8  
Gabon                                    8.9  
Germany                                 11.3  
Lithuania                               12.9  
...                                      ...  
Kuwait                                   0.0  
Iran                                     0.0  
Cook Islands                             5.9  
Bangladesh                               0.0  
Libya                                    0.0  

[193 rows x 4 columns]

In [127]:
#country consuming maximum amount of beer
beer_quantity=df.groupby(['country']).sum()
beer_quantity.sort_values('beer_servings',ascending=False)

beer_servings  spirit_servings  wine_servings  \
country                                                         
Namibia                   376                3              1   
Czech Republic            361              170            134   
Gabon                     347               98             59   
Germany                   346              117            175   
Lithuania                 343              244             56   
...                       ...              ...            ...   
Kuwait                      0                0              0   
Iran                        0                0              0   
Cook Islands                0              254             74   
Bangladesh                  0                0              0   
Libya                       0                0              0   

                total_litres_of_pure_alcohol  
country                                       
Namibia                                  6.8  
Czech Republic                          11.8  
Gabon                                    8.9  
Germany                                 11.3  
Lithuania                               12.9  
...                                      ...  
Kuwait                                   0.0  
Iran                                     0.0  
Cook Islands                             5.9  
Bangladesh                               0.0  
Libya                                    0.0  

[193 rows x 4 columns]

In [129]:
#country consuming max wine
max_wine=df.groupby(['country']).sum().sort_values('wine_servings',ascending=False)
max_wine

beer_servings  spirit_servings  wine_servings  \
country                                                       
France                  127              151            370   
Portugal                194               67            339   
Andorra                 245              138            312   
Switzerland             185              100            280   
Denmark                 224               81            278   
...                     ...              ...            ...   
Rwanda                   43                2              0   
San Marino                0                0              0   
Saudi Arabia              0                5              0   
Somalia                   0                0              0   
Libya                     0                0              0   

              total_litres_of_pure_alcohol  
country                                     
France                                11.8  
Portugal                              11.0  
Andorra                               12.4  
Switzerland                           10.2  
Denmark                               10.4  
...                                    ...  
Rwanda                                 6.8  
San Marino                             0.0  
Saudi Arabia                           0.1  
Somalia                                0.0  
Libya                                  0.0  

[193 rows x 4 columns]

In [145]:
df=pd.read_csv(r"C:\Users\Aswini\Downloads\Data Ananlysis-20210717T051626Z-001\Data Ananlysis\US_Crime_Rates_1960_2014.csv",sep=",")

In [146]:
df.head()

Year  Population    Total  Violent  Property  Murder  Forcible_Rape  \
0  1960   179323175  3384200   288460   3095700    9110          17190   
1  1961   182992000  3488000   289390   3198600    8740          17220   
2  1962   185771000  3752200   301510   3450700    8530          17550   
3  1963   188483000  4109500   316970   3792500    8640          17650   
4  1964   191141000  4564600   364220   4200400    9360          21420   

   Robbery  Aggravated_assault  Burglary  Larceny_Theft  Vehicle_Theft  
0   107840              154320    912100        1855400         328200  
1   106670              156760    949600        1913000         336000  
2   110860              164570    994300        2089600         366800  
3   116470              174210   1086400        2297800         408300  
4   130390              203050   1213200        2514400         472800

In [133]:
df.shape

(55, 12)

In [134]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55 entries, 0 to 54
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   Year                55 non-null     int64
 1   Population          55 non-null     int64
 2   Total               55 non-null     int64
 3   Violent             55 non-null     int64
 4   Property            55 non-null     int64
 5   Murder              55 non-null     int64
 6   Forcible_Rape       55 non-null     int64
 7   Robbery             55 non-null     int64
 8   Aggravated_assault  55 non-null     int64
 9   Burglary            55 non-null     int64
 10  Larceny_Theft       55 non-null     int64
 11  Vehicle_Theft       55 non-null     int64
dtypes: int64(12)
memory usage: 5.3 KB


In [149]:
#Convert the type of the column Year to datetime64
df['Year']=pd.to_datetime(df['Year'],format='%Y')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55 entries, 0 to 54
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Year                55 non-null     datetime64[ns]
 1   Population          55 non-null     int64         
 2   Total               55 non-null     int64         
 3   Violent             55 non-null     int64         
 4   Property            55 non-null     int64         
 5   Murder              55 non-null     int64         
 6   Forcible_Rape       55 non-null     int64         
 7   Robbery             55 non-null     int64         
 8   Aggravated_assault  55 non-null     int64         
 9   Burglary            55 non-null     int64         
 10  Larceny_Theft       55 non-null     int64         
 11  Vehicle_Theft       55 non-null     int64         
dtypes: datetime64[ns](1), int64(11)
memory usage: 5.3 KB


In [150]:
df.set_index(['Year'],inplace=True)

In [141]:
df.head()

Population    Total  Violent  Property  Murder  Forcible_Rape  \
Year                                                                        
1960-01-01   179323175  3384200   288460   3095700    9110          17190   
1961-01-01   182992000  3488000   289390   3198600    8740          17220   
1962-01-01   185771000  3752200   301510   3450700    8530          17550   
1963-01-01   188483000  4109500   316970   3792500    8640          17650   
1964-01-01   191141000  4564600   364220   4200400    9360          21420   

            Robbery  Aggravated_assault  Burglary  Larceny_Theft  \
Year                                                               
1960-01-01   107840              154320    912100        1855400   
1961-01-01   106670              156760    949600        1913000   
1962-01-01   110860              164570    994300        2089600   
1963-01-01   116470              174210   1086400        2297800   
1964-01-01   130390              203050   1213200        2514400   

            Vehicle_Theft  
Year                       
1960-01-01         328200  
1961-01-01         336000  
1962-01-01         366800  
1963-01-01         408300  
1964-01-01         472800

In [143]:
#Group the year by decades and sum the values
df=df.resample('10AS').sum()
df.head()

Population      Total   Violent   Property  Murder  Forcible_Rape  \
Year                                                                            
1960-01-01  1915053175   49295900   4134930   45160900  106180         236720   
1970-01-01  2121193298  100991600   9607930   91383800  192230         554570   
1980-01-01  2371370069  131123369  14074328  117048900  206439         865639   
1990-01-01  2612825258  136582146  17527048  119053499  211664         998827   
2000-01-01  2947969117  115012044  13968056  100944369  163068         922499   

            Robbery  Aggravated_assault  Burglary  Larceny_Theft  \
Year                                                               
1960-01-01  1633510             2158520  13321100       26547700   
1970-01-01  4159020             4702120  28486000       53157800   
1980-01-01  5383109             7619130  33073494       72040253   
1990-01-01  5748930            10568963  26750015       77679366   
2000-01-01  4230366             8652124  21565176       67970291   

            Vehicle_Theft  
Year                       
1960-01-01        5292100  
1970-01-01        9739900  
1980-01-01       11935411  
1990-01-01       14624418  
2000-01-01       11412834

In [151]:
#Select just the 'Murder' and 'Robbery' columns from the DataFrame df
murder_robbery=df[['Murder','Robbery']]
murder_robbery.head()

Murder  Robbery
Year                       
1960-01-01    9110   107840
1961-01-01    8740   106670
1962-01-01    8530   110860
1963-01-01    8640   116470
1964-01-01    9360   130390

In [153]:
#Select the data in rows [3, 4, 8] and in columns ['Murder', 'Robbery']
murder_robbery.iloc[[3,4,8]]

Murder  Robbery
Year                       
1963-01-01    8640   116470
1964-01-01    9360   130390
1968-01-01   13800   262840

In [154]:
df[['Murder','Robbery']].iloc[[3,4,8]]

Murder  Robbery
Year                       
1963-01-01    8640   116470
1964-01-01    9360   130390
1968-01-01   13800   262840

In [156]:
#Select only the rows where the number of murder is greater than 24,000
murder24k=df[df['Murder']>24000]
murder24k

Population     Total  Violent  Property  Murder  Forcible_Rape  \
Year                                                                         
1991-01-01   252177000  14872900  1911770  12961100   24700         106590   
1993-01-01   257908000  14144800  1926020  12218800   24530         106010   

            Robbery  Aggravated_assault  Burglary  Larceny_Theft  \
Year                                                               
1991-01-01   687730             1092740   3157200        8142200   
1993-01-01   659870             1135610   2834800        7820900   

            Vehicle_Theft  
Year                       
1991-01-01        1661700  
1993-01-01        1563100

In [ ]:
#Select the rows the murder is between 20k and 24k (inclusive)

In [160]:
df[df['Murder'].between(20000 ,24000)]

Population     Total  Violent  Property  Murder  Forcible_Rape  \
Year                                                                         
1974-01-01   211392000  10253400   974720   9278700   20710          55400   
1975-01-01   213124000  11292400  1039710  10252700   20510          56090   
1979-01-01   220099000  12249500  1208030  11041500   21460          76390   
1980-01-01   225349264  13408300  1344520  12063700   23040          82990   
1981-01-01   229146000  13423800  1361820  12061900   22520          82500   
1982-01-01   231534000  12974400  1322390  11652000   21010          78770   
1986-01-01   240132887  13211869  1489169  11722700   20613          91459   
1987-01-01   242282918  13508700  1483999  12024700   20096          91110   
1988-01-01   245807000  13923100  1566220  12356900   20680          92490   
1989-01-01   248239000  14251400  1646040  12605400   21500          94500   
1990-01-01   248709873  14475600  1820130  12655500   23440         102560   
1992-01-01   255082000  14438200  1932270  12505900   23760         109060   
1994-01-01   260341000  13989500  1857670  12131900   23330         102220   
1995-01-01   262755000  13862700  1798790  12063900   21610          97470   

            Robbery  Aggravated_assault  Burglary  Larceny_Theft  \
Year                                                               
1974-01-01   442400              456210   3039200        5262500   
1975-01-01   470500              492620   3265300        5977700   
1979-01-01   480700              629480   3327700        6601000   
1980-01-01   565840              672650   3795200        7136900   
1981-01-01   592910              663900   3779700        7194400   
1982-01-01   553130              669480   3447100        7142500   
1986-01-01   542775              834322   3241410        7257153   
1987-01-01   517704              855088   3236184        7499900   
1988-01-01   542970              910090   3218100        7705900   
1989-01-01   578330              951710   3168200        7872400   
1990-01-01   639270             1054860   3073900        7945700   
1992-01-01   672480             1126970   2979900        7915200   
1994-01-01   618950             1113180   2712800        7879800   
1995-01-01   580510             1099210   2593800        7997700   

            Vehicle_Theft  
Year                       
1974-01-01         977100  
1975-01-01        1009600  
1979-01-01        1112800  
1980-01-01        1131700  
1981-01-01        1087800  
1982-01-01        1062400  
1986-01-01        1224137  
1987-01-01        1288674  
1988-01-01        1432900  
1989-01-01        1564800  
1990-01-01        1635900  
1992-01-01        1610800  
1994-01-01        1539300  
1995-01-01        1472400

In [ ]:
#Calculate the mean murder for each different year in df

In [162]:
df.groupby('Year')['Murder'].mean()

Year
1960-01-01     9110.0
1961-01-01     8740.0
1962-01-01     8530.0
1963-01-01     8640.0
1964-01-01     9360.0
1965-01-01     9960.0
1966-01-01    11040.0
1967-01-01    12240.0
1968-01-01    13800.0
1969-01-01    14760.0
1970-01-01    16000.0
1971-01-01    17780.0
1972-01-01    18670.0
1973-01-01    19640.0
1974-01-01    20710.0
1975-01-01    20510.0
1976-01-01    18780.0
1977-01-01    19120.0
1978-01-01    19560.0
1979-01-01    21460.0
1980-01-01    23040.0
1981-01-01    22520.0
1982-01-01    21010.0
1983-01-01    19310.0
1984-01-01    18690.0
1985-01-01    18980.0
1986-01-01    20613.0
1987-01-01    20096.0
1988-01-01    20680.0
1989-01-01    21500.0
1990-01-01    23440.0
1991-01-01    24700.0
1992-01-01    23760.0
1993-01-01    24530.0
1994-01-01    23330.0
1995-01-01    21610.0
1996-01-01    19650.0
1997-01-01    18208.0
1998-01-01    16914.0
1999-01-01    15522.0
2000-01-01    15586.0
2001-01-01    16037.0
2002-01-01    16229.0
2003-01-01    16528.0
2004-01-01    16148.0
2005-

In [163]:
#Sort df first by the values in the 'Murder' in decending order,
#then by the value in the 'Violent' column in ascending order.

In [166]:
df=df.sort_values(['Murder'],ascending=False)
df.head()

Population     Total  Violent  Property  Murder  Forcible_Rape  \
Year                                                                         
1991-01-01   252177000  14872900  1911770  12961100   24700         106590   
1993-01-01   257908000  14144800  1926020  12218800   24530         106010   
1992-01-01   255082000  14438200  1932270  12505900   23760         109060   
1990-01-01   248709873  14475600  1820130  12655500   23440         102560   
1994-01-01   260341000  13989500  1857670  12131900   23330         102220   

            Robbery  Aggravated_assault  Burglary  Larceny_Theft  \
Year                                                               
1991-01-01   687730             1092740   3157200        8142200   
1993-01-01   659870             1135610   2834800        7820900   
1992-01-01   672480             1126970   2979900        7915200   
1990-01-01   639270             1054860   3073900        7945700   
1994-01-01   618950             1113180   2712800        7879800   

            Vehicle_Theft  
Year                       
1991-01-01        1661700  
1993-01-01        1563100  
1992-01-01        1610800  
1990-01-01        1635900  
1994-01-01        1539300

In [167]:
df.sort_values(["Violent"],ascending=True)

Population     Total  Violent  Property  Murder  Forcible_Rape  \
Year                                                                         
1960-01-01   179323175   3384200   288460   3095700    9110          17190   
1961-01-01   182992000   3488000   289390   3198600    8740          17220   
1962-01-01   185771000   3752200   301510   3450700    8530          17550   
1963-01-01   188483000   4109500   316970   3792500    8640          17650   
1964-01-01   191141000   4564600   364220   4200400    9360          21420   
1965-01-01   193526000   4739400   387390   4352000    9960          23410   
1966-01-01   195576000   5223500   430180   4793300   11040          25820   
1967-01-01   197457000   5903400   499930   5403500   12240          27620   
1968-01-01   199399000   6720200   595010   6125200   13800          31670   
1969-01-01   201385000   7410900   661870   6749000   14760          37170   
1970-01-01   203235298   8098000   738820   7359200   16000          37990   
1971-01-01   206212000   8588200   816500   7771700   17780          42260   
1972-01-01   208230000   8248800   834900   7413900   18670          46850   
1973-01-01   209851000   8718100   875910   7842200   19640          51400   
1974-01-01   211392000  10253400   974720   9278700   20710          55400   
1976-01-01   214659000  11349700  1004210  10345500   18780          57080   
1977-01-01   216332000  10984500  1029580   9955000   19120          63500   
1975-01-01   213124000  11292400  1039710  10252700   20510          56090   
1978-01-01   218059000  11209000  1085550  10123400   19560          67610   
2014-01-01   318857056   9475816  1197987   8277829   14249          84041   
2013-01-01   316497531   9850445  1199684   8650761   14319          82109   
2011-01-01   311587816  10258774  1206031   9052743   14661          84175   
1979-01-01   220099000  12249500  1208030  11041500   21460          76390   
2012-01-01   313873685  10219059  1217067   9001992   14866          85141   
2010-01-01   309330219  10363873  1251248   9112625   14772          85593   
1983-01-01   233981000  12108600  1258090  10850500   19310          78920   
1984-01-01   236158000  11881800  1273280  10608500   18690          84230   
1982-01-01   231534000  12974400  1322390  11652000   21010          78770   
2009-01-01   307006550  10762956  1325896   9337060   15399          89241   
1985-01-01   238740000  12431400  1328800  11102600   18980          88670   
1980-01-01   225349264  13408300  1344520  12063700   23040          82990   
2004-01-01   293656842  11679474  1360088  10319386   16148          95089   
1981-01-01   229146000  13423800  1361820  12061900   22520          82500   
2003-01-01   290690788  11826538  1383676  10442862   16528          93883   
2005-01-01   296507061  11565499  1390745  10174754   16740          94347   
2008-01-01   304374846  11160543  1392628   9767915   16442          90479   
2007-01-01   301621157  11251828  1408337   9843481   16929          90427   
2006-01-01   299398484  11401511  1418043   9983568   17030          92757   
2002-01-01   287973924  11878954  1423677  10455277   16229          95235   
2000-01-01   281421906  11608072  1425486  10182586   15586          90178   
1999-01-01   272690813  11634378  1426044  10208334   15522          89411   
2001-01-01   285317559  11876669  1439480  10437480   16037          90863   
1987-01-01   242282918  13508700  1483999  12024700   20096          91110   
1986-01-01   240132887  13211869  1489169  11722700   20613          91459   
1998-01-01   270296000  12475634  1531044  10944590   16914          93103   
1988-01-01   245807000  13923100  1566220  12356900   20680          92490   
1997-01-01   267637000  13194571  1634770  11558175   18208          96153   
1989-01-01   248239000  14251400  1646040  12605400   21500          94500   
1996-01-01   265228572  13493863  1688540  11805300   19650          96250   
1995-01-01   262755000  13862700  

In [168]:
pwd()

'C:\\Users\\Aswini\\Downloads\\Data Ananlysis-20210717T051626Z-001\\Data Ananlysis'